### PR TITLE
Update Cargo submodule

### PR DESCRIPTION
**Pull request by alexcrichton**
Hopefully a routine update...

### CODE REVIEW

File: src/Cargo.lock [hunk 1]
```diff
@@ -2894,15 +2945,17 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 "checksum quote 0.3.15 (registry+https://github.com/rust-lang/crates.io-index)" = "7a6e920b65c65f10b2ae65c831a81a073a89edd28c7cce89475bff467ab4167a"
 "checksum racer 2.0.12 (registry+https://github.com/rust-lang/crates.io-index)" = "034f1c4528581c40a60e96875467c03315868084e08ff4ceb46a00f7be3b16b4"
 "checksum radix_trie 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)" = "211c49b6a9995cac0fd1dd9ca60b42cf3a51e151a12eb954b3a9e75513426ee8"
-"checksum rand 0.3.20 (registry+https://github.com/rust-lang/crates.io-index)" = "512870020642bb8c221bf68baa1b2573da814f6ccfe5c9699b1c303047abe9b1"
+"checksum rand 0.3.22 (registry+https://github.com/rust-lang/crates.io-index)" = "15a732abf9d20f0ad8eeb6f909bf6868722d9a06e1e50802b6a70351f40b4eb1"
+"checksum rand 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)" = "eba5f8cb59cc50ed56be8880a5c7b496bfd9bd26394e176bc67884094145c2c5"
```
Comment (Mark-Simulacrum) [inline src/Cargo.lock]: This'll break some of our tests IIRC so we'll need to not do this until we can get rid of one of the rands or remove `extern crate rand` from the tests.

**Review by Mark-Simulacrum — APPROVED** [review]
r=me if you have a solution for the problem I've noted

**Comment by Manishearth** [comment]
@bors p=2

prioritizing risky-to-rollup PRs so they get tested first

**Comment by kennytm** [comment]
@bors r- 

Cargo.lock is not up-to-date.

<details>

```
[00:02:11]     Updating registry `https://github.com/rust-lang/crates.io-index`
[00:02:11] error: the lock file needs to be updated but --locked was passed to prevent this
[00:02:11] failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml --locked
[00:02:11] Build completed unsuccessfully in 0:00:00
[00:02:11] make: *** [prepare] Error 1
[00:02:11] Makefile:76: recipe for target 'prepare' failed
[00:02:11] The command has failed after 5 attempts.
```

</details>

**Comment by alexcrichton** [comment]
@bors: r-

er, not actually ready yet

**Comment by alexcrichton** [comment]
@bors: retry

I'd rather #48125 go first

**Comment by Manishearth** [comment]
```
[01:05:02] travis_time:end:stage2-rustdoc:start=1520201952880828080,finish=1520202085790167062,duration=132909338982
[0K
[01:05:02] travis_fold:start:test_stage2-core
[0Ktravis_time:start:test_stage2-core
[0KTesting core stage2 (x86_64-unknown-linux-gnu -> wasm32-unknown-unknown)
[01:05:02] Libstd was built without `wasm_syscall` feature enabled: test output may not be visible.
[01:05:02] [m[m[32m[1m   Compiling[m rand v0.4.2
[01:05:04] [m[m[32m[1m   Compiling[m core v0.0.0 (file:///checkout/src/libcore)
[01:06:14] [m[m[32m[1m    Finished[m release [optimized] target(s) in 72.60 secs
[01:06:14] [m[m[32m[1m     Running[m build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps/coretests-c4daadc4973c9f7a.wasm
[01:06:15] RuntimeError: unreachable
[01:06:15]     at __rust_start_panic (wasm-function[6799]:1)
[01:06:15]     at rust_panic.llvm.13901392069640985989 (wasm-function[5796]:26)
[01:06:15]     at _ZN3std9panicking20rust_panic_with_hook17hc0b4bf5fdcf60eccE (wasm-function[5795]:533)
[01:06:15]     at _ZN3std9panicking11begin_panic17hdea0c27d8fe149a0E (wasm-function[3335]:74)
[01:06:15]     at _ZN4rand6jitter8platform10get_nstime17hcdf8754920a1bcccE (wasm-function[3500]:15)
[01:06:15]     at _ZN4rand10thread_rng14THREAD_RNG_KEY6__init17h1a34bac544788ea7E (wasm-function[3467]:82)
[01:06:15]     at _ZN46_$LT$std..thread..local..LocalKey$LT$T$GT$$GT$4with17hedcf73a54ae0e030E (wasm-function[3484]:46)
[01:06:15]     at _ZN4rand10thread_rng17hd01539dec0e64c14E (wasm-function[3461]:7)
[01:06:15]     at _ZN9coretests3num7flt2dec6random33exact_f32_random_equivalence_test17h95de4ae404df59e0E (wasm-function[385]:174)
[01:06:15]     at _ZN4core3ops8function6FnOnce9call_once17h0dfb3053b1b21721E (wasm-function[1857]:21)
[01:06:15]     at _ZN4test28__rust_begin_short_backtrace17h5fc66a0063efa6d2E (wasm-function[3546]:3)
[01:06:15]     at _ZN50_$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$8call_box17h64b4c74fbd9cab25E (wasm-function[3545]:6)
[01:06:15]     at _ZN3std9panicking3try7do_call17hc85a05c2d4b1ea80E.llvm.3533770085478341910 (wasm-function[4280]:14)
[01:06:15]     at __rust_maybe_catch_panic (wasm-function[6798]:5)
[01:06:15]     at _ZN3std9panicking3try17ha66ff1436d4092ceE (wasm-function[4279]:74)
[01:06:15]     at _ZN4test8run_test14run_test_inner17h8656dcd2bf469cb3E (wasm-function[3595]:784)
[01:06:15]     at _ZN4test8run_test17hbd3fd90bfbd5c34fE (wasm-function[3589]:1022)
[01:06:15]     at _ZN4test17run_tests_console17h1574a6cb96a29259E (wasm-function[3580]:4185)
[01:06:15]     at _ZN4test9test_main17h1fc9ace9af72838bE (wasm-function[3577]:489)
[01:06:15]     at _ZN4test16test_main_static17hc811569303827aa6E (wasm-function[3581]:533)
[01:06:15] [m[m[31m[1merror:[m test failed, to rerun pass '--test coretests'
[01:06:15] 
[01:06:15] 
[01:06:15] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "wasm32-unknown-unknown" "-j" "4" "--release" "--locked" "--color" "always" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "-p" "core" "--"
[01:06:15] expected success, got: exit code: 101
[01:06:15] 
[01:06:15] 
[01:06:15] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test --target wasm32-unknown-unknown src/test/ui src/test/run-pass src/test/compile-fail src/test/parse-fail src/test/mir-opt src/test/codegen-units src/libcore src/libstd_unicode/
[01:06:15] Build completed unsuccessfully in 1:03:36


```

**Comment by Manishearth** [comment]
```
Testing core stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
[01:18:20]    Compiling core v0.0.0 (file:///checkout/src/libcore)
[01:18:20] error: an inner attribute is not permitted following an outer attribute
[01:18:20]    --> libcore/../libcore/tests/slice.rs:489:3
[01:18:20]     |
[01:18:20] 489 | #![cfg(not(target_arch = "wasm32"))]
[01:18:20]     |   ^
[01:18:20]     |
[01:18:20]     = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files. Outer attributes, like `#[test]`, annotate the item following them.
[01:18:20] 
[01:18:34] error: aborting due to previous error
[01:18:34] 
[01:18:34] error: Could not compile `core`.
[01:18:34] 
[01:18:34] To learn more, run the command again with --verbose.
[01:18:34] 
[01:18:34] 
[01:18:34] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "x86_64-unknown-linux-gnu" "--release" "--locked" "--color" "always" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "-p" "core" "--" "--quiet"
[01:18:34] expected success, got: exit code: 101
[01:18:34] 
[01:18:34] 
[01:18:34] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:18:34] Build completed unsuccessfully in 0:23:52
[01:18:34] make: *** [check] Error 1
[01:18:34] Makefile:52: recipe for target 'check' failed
```

@bors r- p=0

Perhaps send this one through try before r+ing?

**Comment by alexcrichton** [comment]
@bors: retry

I'd prefer if #48642 landed first

**Comment by alexcrichton** [comment]
@bors: retry

github probably isn't down

**Comment by alexcrichton** [comment]
@bors: retry

* prioritizing rollup with this in it

**Comment by alexcrichton** [comment]
@bors: retry

* prioritizing rollup

**Comment by alexcrichton** [comment]
@bors: r-

gotta [figure this out](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.6585)

**Comment by alexcrichton** [comment]
Should be fixed by https://github.com/alexcrichton/git2-rs/commit/04c1b914a2e96a50677c54ab3ee2233221c1ebd2

@bors: r=Mark-Simulacrum

**Comment by kennytm** [comment]
<details><summary>Unused imports.</summary>

```
[01:16:36] Libstd was built without `wasm_syscall` feature enabled: test output may not be visible.
[01:16:37]    Compiling rand v0.4.2
[01:16:39]    Compiling core v0.0.0 (file:///checkout/src/libcore)
[01:16:41] error: unused imports: `Equal`, `Greater`, `Less`
[01:16:41]   --> libcore/../libcore/tests/slice.rs:12:27
[01:16:41]    |
[01:16:41] 12 | use core::cmp::Ordering::{Equal, Greater, Less};
[01:16:41]    |                           ^^^^^  ^^^^^^^  ^^^^
[01:16:41]    |
[01:16:41] note: lint level defined here
[01:16:41]   --> libcore/../libcore/tests/lib.rs:11:9
[01:16:41]    |
[01:16:41] 11 | #![deny(warnings)]
[01:16:41]    |         ^^^^^^^^
[01:16:41]    = note: #[deny(unused_imports)] implied by #[deny(warnings)]
[01:16:41]
[01:16:41] error: unused import: `core::slice::heapsort`
[01:16:41]   --> libcore/../libcore/tests/slice.rs:13:5
[01:16:41]    |
[01:16:41] 13 | use core::slice::heapsort;
[01:16:41]    |     ^^^^^^^^^^^^^^^^^^^^^
[01:16:41]
[01:16:41] error: unused imports: `Rng`, `XorShiftRng`
[01:16:41]   --> libcore/../libcore/tests/slice.rs:15:12
[01:16:41]    |
[01:16:41] 15 | use rand::{Rng, XorShiftRng};
[01:16:41]    |            ^^^  ^^^^^^^^^^^
[01:16:41]
[01:16:55] error: aborting due to 3 previous errors
[01:16:55]
[01:16:55] error: Could not compile `core`.
```

</details>

**Comment by kennytm** [comment]
`asmjs` failed at `slice::panic_safe`.

<details>

```
[02:22:46] failures:
[02:22:46] 
[02:22:46] ---- slice::panic_safe stdout ----
[02:22:46] 	thread 'main' panicked at 'explicit panic', libstd/sys/unix/thread.rs:48:5
[02:22:46] 
[02:22:46] 
[02:22:46] failures:
[02:22:46]     slice::panic_safe
[02:22:46] 
[02:22:46] test result: FAILED. 479 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out
[02:22:46] 
[02:22:46] [m[m[31m[1merror:[m test failed, to rerun pass '--test collectionstests'
```

</details>

**Pull request merged**